code/modules/mob/living/bot/cleanbot.dm

```diff
@@ -101,25 +107,23 @@
 		patrol_path = list()
 		return
 
-	while (searching)
-		for (current_tile = 0, current_tile <= maximum_search_range, current_tile++)
-			for (cleanable_type in view(current_tile, src))
-				if (!(cleanable_type in ignorelist))
-					for (target_type in target_types)
-						if (istype(cleanable_type, target_type))
+	var/found_spot
+	search_for:
+		for(var/i=0, i <= maximum_search_range, i++)
+			clean_for:
+				for(var/obj/effect/decal/cleanable/D in view(i, src))
+					if((D in ignorelist)) continue clean_for
+					for(var/T in target_types)
+						if(istype(D, T))
 							patrol_path = list()
-							target      = cleanable_type
-							found_spot  = handle_target()
-
+							target = D
+							found_spot = handle_target()
 							if (found_spot)
-								searching = 0;
+								break search_for
 							else
-								if (target == null) //handles if path can not be created
-									searching = 0
+								target = null
+								continue // no need to check the other types
```

Comment:
Whoops, I think that comment was leftover from stuff I removed from the Baycode I copied in, you're right in what you said. It's meant to continue the type check, not the view() check (which will continue on its own if no break/return is reached), the break above it will break out of all loops which will prevent all further matches. That continue will keep looking through the valid types. However, I can probably just do an if((D.type in target_types)) in place of that whole loop and be fine. I'll update the code for that (surprised I didn't catch it sooner) and make sure it works as expected.